src/phase0/state_transition/mod.rs

```diff
@@ -60,6 +60,18 @@ pub enum Error {
         previous: Epoch,
         current: Epoch,
     },
+    #[error("invalid block header: {0}")]
+    InvalidHeader(InvalidHeader),
+}
+
+#[derive(Debug, Error)]
+pub enum InvalidHeader {
```

Comment:
this does increase the verbosity but we will be happier later when there are other types of `Header`s floating around (e.g. a `ShardHeader`).
```suggestion
pub enum InvalidBeaconBlockHeader {
```